switchmap/server/db/table/metrics.py

```
@@ -0,0 +1,89 @@
+        None
+
+    """
+    if not isinstance(rows, list):
+        rows = [rows]
+
+    inserts = []
+    for row in rows:
+
+        # Validate/encode hostname (VARBINARY NOT NULL)
+        if getattr(row, "hostname", None) is None:
+            raise ValueError("hostname is required for DeviceMetricsHistory")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)